### DISCUSSION TITLE
rebaseWhen new value for Automerge.

### DISCUSSION THREAD

**Discussion by PhilipAbed** [renovatebot/renovate#31398] [Suggest an Idea]
### Tell us more.

currently https://docs.renovatebot.com/configuration-options/#rebasewhen
shows:
`auto`: Renovate will autodetect the best setting. It will use behind-base-branch if configured to automerge or repository has been set to require PRs to be up to date. Otherwise, conflicted will be used instead
`never`: Renovate will never rebase the branch or update it unless manually requested
`conflicted`: Renovate will rebase only if the branch is conflicted
`behind-base-branch`: Renovate will rebase whenever the branch falls 1 or more commit behind its base branch

I would like to add an option value of ~`automerge`~ `automerging` : 
Renovate will use `behind-base-branch` if configured to automerge, ~and repository has been set to require PRs to be up to date~ . Otherwise `never`.

**Comment by nabeelsaabna** [comment]
this new option is aimed to help with reducing rebase action done by Renovate ( e.g. `rebaseWhen = never` ) without blocking automerge behavior that needs the branch to be "not conflicted" and sometime "not behind-base-branch"

`rebaseWhen = automerge`:
1. will only be applied when `automerge` is enabled :white_check_mark: 
2. and repository has been set to require PRs to be up to date ?? I believe this is an additional optimization

**Reply by rarkins** [reply]
I'm not sure about point 2. I've been thinking that "always rebase automerging PRs" was an opinionated non-configurable behavior I added long ago and maybe there should be the possibility to opt out of it. Eg we probably should in this repo and let the merge queue take over responsibility for testing against latest commit

**Comment by rarkins** [comment]
Can we name it "automerging" to differentiate it a bit more from the existing automerge setting?

**Reply by PhilipAbed** [reply]
yes, i changed description, should we turn this into an issue?

**Reply by rarkins** [reply]
Yes but wait a day for another feedback.

**Comment by PhilipAbed** [comment]
i also noticed these cases:
https://github.com/renovatebot/renovate/blob/b0a0a714cd576cd60d6dce00bc095f3c9c0d7bc1/lib/workers/repository/update/branch/reuse.ts#L45C5-L55C48

should i handle them the same way as auto in case of Automerging? or just `never` regardless of their values?

**Reply by rarkins** [reply]
Skip the second clause (force rebase) but honor the third one (keep updating label)

**Comment by PhilipAbed** [comment]
https://github.com/renovatebot/renovate/pull/31527
pr merged.
